content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,351 @@
+
+</TopBlock>
+
+## What is MongoDB?
+
+MongoDB is a NoSQL database that stores data in [BSON](https://bsonspec.org/) format, a JSON-like document format designed for storing data in key-value pairs.
+
+MongoDB is commonly used in JavaScript application development because the document model maps easily to objects in application code, and there is built in support for high availability and horizontal scaling.
+
+MongoDB stores data in collections that do not need a schema to be defined in advance, as you would need to do with tables in a relational database. The structure of each collection can also be changed over time. This flexibility can allow rapid iteration of your data model, but it does mean that there are a number of differences when using Prisma to work with your MongoDB database.
+
+## Commonalities with other database providers
+
+Some aspects of using Prisma with MongoDB are the same as when using Prisma with a relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/concepts/components/prisma-schema)
+- connect to your database, using the [`mongodb` database connector](/concepts/database-connectors/mongodb)
+- use [Introspection](/concepts/components/introspection) for existing projects if you already have a MongoDB database
+- use [`db push`](/concepts/components/prisma-migrate/db-push) to push changes in your schema to the database
+- use [Prisma Client](/concepts/components/prisma-client) in your application to query your database
+
+## Differences to consider
+
+MongoDB's document-based structure and flexible schemas means that using Prisma with MongoDB differs from using it with a relational database in a number of ways. These are some areas where there are differences that you need to be aware of:
+
+- **Defining IDs**: MongoDB documents have an `_id` field that contain an [ObjectID](https://www.mongodb.com/docs/manual/reference/bson-types/#std-label-objectid). This needs to be mapped to a Prisma field. For more information, see [Defining IDs in MongoDB](/concepts/components/prisma-schema/data-model#defining-ids-in-mongodb).
+
+- **Migrating old data to new schemas**: In relational databases, all your data must match your schema. If you change the type of a particular field in your schema when you migrate, all the data must also be updated to match. In contrast, MongoDB does not enforce any particular schema, so you need to take care when migrating. For more information, see [How to migrate old data to new schemas](#how-to-migrate-old-data-to-new-schemas).
+
+- **Introspection**: When you introspect an existing database, you will get a schema with no relations and will need to add the missing relations in manually. For more information, see [How to add in missing relations after Introspection](#how-to-add-in-missing-relations-after-introspection).
```

Comment:
I'd lean towards naming this `Introspection and Prisma relations`.

Reply:
This is relevant, because MongoDB does not haave a concept of relations (via foreign keys). Only if you have data that has a "foreign key" like column, that has the same value as the ID field of another collection you can model it like that and Prisma lets you emulate relations on that that behave like real ones.

Nuance, but important.

Reply:
Changed subsection title and clarified this in text of "how to..." section